src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/ProcessHmcMessageService.java

```diff
@@ -0,0 +1,120 @@
+package uk.gov.hmcts.reform.sscs.service.hmc.topic;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.exception.CaseException;
+import uk.gov.hmcts.reform.sscs.exception.MessageProcessingException;
+import uk.gov.hmcts.reform.sscs.model.hmc.message.HmcMessage;
+import uk.gov.hmcts.reform.sscs.model.hmc.reference.HmcStatus;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingGetResponse;
+import uk.gov.hmcts.reform.sscs.service.CcdCaseService;
+import uk.gov.hmcts.reform.sscs.service.HmcHearingApiService;
+
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.reform.sscs.model.hmc.reference.HmcStatus.CANCELLED;
+import static uk.gov.hmcts.reform.sscs.model.hmc.reference.HmcStatus.EXCEPTION;
+import static uk.gov.hmcts.reform.sscs.model.hmc.reference.HmcStatus.LISTED;
+import static uk.gov.hmcts.reform.sscs.model.hmc.reference.HmcStatus.UPDATE_SUBMITTED;
+import static uk.gov.hmcts.reform.sscs.model.hmc.reference.ListingStatus.FIXED;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ProcessHmcMessageService {
+
+    private static final String REQUEST_CANCELLED = "Cancelled";
+
+    @Value("${sscs.serviceCode}")
+    private String sscsServiceCode;
+
```

Comment:
Medium: Better to pass this in constructor so easier to unit test

Reply:
It is passed in the constructor using RequiredArgsConstructor, InjectMocks is used in the unit tests.